lib/class-wp-theme-json-gutenberg.php

```diff
@@ -1059,7 +1076,7 @@ private function get_block_classes( $style_nodes ) {
 			}
 
 			if ( self::ROOT_BLOCK_SELECTOR === $selector ) {
-				$block_rules .= 'body { margin: 0; }';
```

Comment:
Avoid applying rule here otherwise it will end up being applied last in the selector order thereby ensuring it takes precedence over rules applied via `theme.json`.